src/Node/index.tsx

```diff
@@ -119,22 +126,27 @@ export default class Node extends React.Component<NodeProps, NodeState> {
   // TODO: needs tests
   renderNodeElement = () => {
     const { data, hierarchyPointNode, renderCustomNodeElement } = this.props;
-    const renderNode = typeof renderCustomNodeElement === 'function' ? renderCustomNodeElement : DefaultNodeElement;
+    const renderNode =
+      typeof renderCustomNodeElement === 'function' ? renderCustomNodeElement : DefaultNodeElement;
     const nodeProps = {
-        hierarchyPointNode: hierarchyPointNode,
-        nodeDatum: data,
-        toggleNode: this.handleNodeToggle,
-        onNodeClick: this.handleOnClick,
-        onNodeMouseOver: this.handleOnMouseOver,
-        onNodeMouseOut: this.handleOnMouseOut,
+      hierarchyPointNode: hierarchyPointNode,
+      nodeDatum: data,
+      toggleNode: this.handleNodeToggle,
+      onNodeClick: this.handleOnClick,
+      onNodeMouseOver: this.handleOnMouseOver,
+      onNodeMouseOut: this.handleOnMouseOut,
     };
 
-    return renderNode(nodeProps)
+    return renderNode(nodeProps);
   };
 
-  handleNodeToggle = () => this.props.onNodeToggle(this.props.data.__rd3t.id);
+  handleNodeToggle = () => {
+    this.setState({ ...this.state, wasClicked: true });
```

Comment:
No need to explicit pass in all of `...this.state`, only the state prop that is changing since `Component.setState` effectively uses `Object.assign()` under the hood to merge the objects:

```suggestion
    this.setState({ wasClicked: true });
```

Reply:
Updated